ddtrace/tracer/remote_config.go

```diff
@@ -80,11 +108,15 @@ func (t *tags) toMap() *map[string]interface{} {
 
 func (t *tracer) dynamicInstrumentationRCUpdate(u remoteconfig.ProductUpdate) map[string]state.ApplyStatus {
 	applyStatus := map[string]state.ApplyStatus{}
-
 	for k, v := range u {
 		log.Debug("Received dynamic instrumentation RC configuration for %s\n", k)
 		applyStatus[k] = state.ApplyStatus{State: state.ApplyStateUnknown}
-		passFullConfiguration(globalconfig.RuntimeID(), k, string(v))
+
+		dynamicInsturmentationRCState.mu.Lock()
+		dynamicInsturmentationRCState.runtimeID = globalconfig.RuntimeID()
+		dynamicInsturmentationRCState.productConfigPath = k
+		dynamicInsturmentationRCState.productConfigContent = string(v)
+		dynamicInsturmentationRCState.mu.Unlock()
 	}
```

Comment:
It's not clear for me how this should work. I see we are overwriting the global state object with every update, but we are calling `passFullConfiguration` every 5s, irrespective of how we receive the updates.

I was imagining the state would be a map holding all configs, not just one.

Reply:
Ah you're right, the config we get from remoteconfig is updates not the full config. Let me fix that.

Reply:
Ok I changed it to use a map and track the state of each probes config, but still passing individual strings to the callback (as they are now). That's because we don't yet properly support maps in Go-DI.